### ISSUE TITLE
Native exception on simulator after upgrading to 2.2.1

### ISSUE THREAD

**Issue by jimfoambox** [bugfender/rn-bugfender#81]
Just recently upgraded our React Native app from 0.68.2 to 0.71.8 and React to 18.2.0, and along with that updated Bugfender to 2.2.1.

When in the iOS simulator (Android has no issues), I am getting a native error every time I make a JS change and metro/the app does a hot reload:

<img width="439" alt="Screen Shot 2023-08-25 at 3 17 26 PM" src="https://github.com/bugfender/rn-bugfender/assets/5530251/410d509b-082d-4f9f-ab13-36c2620b584d">
<img width="384" alt="Screen Shot 2023-08-25 at 3 21 26 PM" src="https://github.com/bugfender/rn-bugfender/assets/5530251/38692543-b9cc-4937-a62f-f3bb838c9ad7">
<img width="373" alt="Screen Shot 2023-08-25 at 3 21 42 PM" src="https://github.com/bugfender/rn-bugfender/assets/5530251/245c7495-48bf-43d0-90ba-070d21e80c73">

We were previously on version 2.1.1

After downgrading back to 2.1.1, the error went away at first, but then started to come back again, leading me now to believe it might be a .71 upgrade compatibility issue?

**Comment by jgimenez** [comment]
Looks like a regression introduced when reimplementing the iOS module as Turbo Native Module. Are you providing a `deviceName` parameter to `Bugfender.init()`? Maybe try without (prefer `setDeviceKey()` instead).

**Comment by jimfoambox** [comment]
@jgimenez  thanks for the reply.

So we experimented today both with providing a deviceName param as well as explicitly setting it to undefined, and found that when it was undefined the error stopped.

Aside from that, our Bugfender.init includes appkey and printToConsole, and we perform setDeviceKey with additional user info after a user logs in.  Would it be optimal if we explicitly set deviceName with the login setDeviceKey instead?

**Comment by jgimenez** [comment]
Yes, the `deviceName` is an old field, and is mostly deprecated now. Maybe we should officially remove it, although we try to avoid breaking changes. The SDK used to get the device name automatically, which sometimes contains the name of the owner (something like "jimfoambox's iPhone"). We provided this field so you could override it with something privacy-preserving, like "Unknown". Due to this historical use, this field can only be set once and the backend will update it rarely. This field is no longer needed because the SDK doesn't gather the device name anymore, for privacy.

So, in short, yes, `setDeviceKey()` is a better way to track which devices belong to which users.

The underlying issue in this exception is that the native Bugfender SDK can only be initialized once, but when using the debugger, it's possible to reload the RN application without reloading the native application container, and the Turbo Native Module is reloaded from scratch, so it tries to initialize the native Bugfender SDK again. We'll check if it's possible to keep this from happening.

**Comment by jimfoambox** [comment]
Roger that, thanks @jgimenez
